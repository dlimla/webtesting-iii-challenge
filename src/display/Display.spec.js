// Test away!
import React from 'react';
import renderer from 'react-test-renderer';
import {render, cleanup } from 'react-testing-library';


import Display from './Display';

describe('<Display />', () => {

    //display render test
    it('renders display without crashing', () => {
        render(<Display />)
    })

    //snapshot!!
    it('should create a match snapshot of display', () => {
        const tree = renderer.create(<Display />);

        expect(tree.toJSON()).toMatchSnapshot();
    })

    //tets default values   
    it('has a default of unlocked and closed so "locked" and "closed" should be false', () => {
        const { getByText } = render(<Display />);
        getByText(/unlocked/i);
        getByText(/open/i);
    })
})