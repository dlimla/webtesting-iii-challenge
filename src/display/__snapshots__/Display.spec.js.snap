// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> should create a match snapshot of display 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;
